app/components/CommunitySelector/DropdownTrigger.tsx

```diff
@@ -0,0 +1,99 @@
+import { ArrowDown } from 'components/icons';
+import useTrigger from 'hooks/useTrigger';
+import React from 'react';
+import { BORDER_PRIMARY } from 'style-constants';
+
+import { singleCommunityColors } from 'utils/communityManagement';
+import { IconMd } from 'components/Icon/IconWithSizes';
+import closeIcon from 'images/closeCircle.svg?external';
+import { styles } from './DropdownTrigger.styled';
+
+const colors = singleCommunityColors();
+
+type DropdownTriggerProps = {
+  selectedLanguages: string[];
+  allLanguages: Language[];
+  removeSelectedLanguage: (removedLanguage: string) => void;
+  placeholder: string;
+};
+
+type Language = {
+  value: string;
+  label: string;
+};
+
+const DropdownTrigger: React.FC<DropdownTriggerProps> = ({
+  selectedLanguages,
+  allLanguages,
+  removeSelectedLanguage,
+  placeholder,
+}): JSX.Element => {
+  const [isOpen, open, close] = useTrigger(false);
+
+  document.addEventListener('click', () => {
```

Comment:
put everything in useEffect and unsubscribe when the component is unmounted

Reply:
done